extensions/dicom-segmentation/src/components/SegmentationSettings/SegmentationSettings.js

```diff
@@ -141,29 +141,36 @@ const SegmentationSettings = ({ configuration, onBack, onChange, servicesManager
             style={{ margin: '0 15px' }}
             label="Tolerance"
             onKeyPress={event => {
-              const validate = string => {
-                let rgx = /[^-.e0-9]+/g;
-                return string.match(rgx);
-              };
+              if (event.key === 'Enter') {
+                const validate = string => {
+                  let rgx = /[^-.e0-9]+/g;
+                  return string.match(rgx);
+                };
 
-              if (validate(event.key)) {
-                event.preventDefault();
+                if (validate(event.key)) {
+                  event.preventDefault();
+                }
+
+                const {
```

Comment:
The message will be displayed only on enter or keypress? I thought this message should be triggered inside onChange. If the problem is multiple calls then you could set up a once function and execute it in the onChange e.g.
```
const once = fn => (...args) => {
  if (!fn) return;
  fn(...args);
  fn = null;
};
```

Reply:
Thanks for the review, Igor!
I have implemented the once function for onChange as described.